omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/FormResource1_8.java

```diff
@@ -194,7 +197,15 @@ protected NeedsPaging<Form> doGetAll(RequestContext context) throws ResponseExce
 	 */
 	@Override
 	protected NeedsPaging<Form> doSearch(RequestContext context) {
-		return new NeedsPaging<Form>(Context.getFormService().getForms(context.getParameter("q"), false), context);
+		String fuzzyName = context.getParameter("q");
+		String retiredQueryParam = context.getParameter("retired");
+		Boolean retired = null;
+		if (retiredQueryParam != null) {
+			retired = Boolean.parseBoolean(retiredQueryParam);
```

Comment:
For retired, use `context.getIncludeAll()`

Reply:
This would include voided/retired in the result set
According to the implementation,

- If `retired` is null (`retired` request parameter is *not* provided) then all forms retired and non-retired are returned.

- If `retired` is not null and evaluates to `true`, only retired forms are retrieved. *?

- If `retired` is not null and evaluates to `false`, only non-retired forms are retrieved.

Reply:
What i mean is that a REST call which has `?includeAll=false` would exclude retired forms if your code just used `context.getIncludeAll()`